src/relay.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 Mateusz Lichota
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <rclcpp/generic_publisher.hpp>
+#include <rclcpp/generic_subscription.hpp>
+
+#include <memory>
+#include <string>
+
+using namespace std::chrono_literals;
+
+const auto LOOP_PERIOD = 100ms;
+
+namespace topic_tools
+{
+
+  class RelayNode : public rclcpp::Node
+  {
+  public:
+    RelayNode(
+      const std::string node_name, 
+      const rclcpp::NodeOptions &options);
+      
+    RelayNode(
```

Comment:
I would argue that, since `NodeOptions` can contain `parameter_overrides` - there's no need for a second specialty constructor. Better to stick with a single way of configuring.

(if it sticks with two constructors) Stylistically I would also avoid a separate `initialize()` function in favor of having one constructor call another, if at all possible. From a usage perspective, it's hard to tell whether it's OK to call `initialize` again after construction, etc.